android/src/com/mapswithme/maps/maplayer/SearchWheel.java

```diff
@@ -122,40 +115,39 @@ public SearchWheel(View frame, @NonNull View.OnClickListener onSearchPressedList
     refreshSearchVisibility();
   }
 
-  private @Nullable View getSearchLayout()
+  private boolean initSearchLayout()
   {
+    if (mSearchLayout != null)
+      return true;
+
+    mSearchLayout = mFrame.findViewById(R.id.search_frame);
     if (mSearchLayout == null)
+      return false;
+
+    DisplayMetrics displayMetrics = new DisplayMetrics();
+    WindowManager windowmanager = (WindowManager) mFrame.getContext().getSystemService(Context.WINDOW_SERVICE);
+    windowmanager.getDefaultDisplay().getMetrics(displayMetrics);
+    // Get available screen height in DP
+    int height =  Math.round(displayMetrics.heightPixels / displayMetrics.density);
+    // If height is less than 400dp, the search wheel in a straight line
+    // In this case, move the pivot for the animation
+    if (height < 400)
     {
-      mSearchLayout = mFrame.findViewById(R.id.search_frame);
-      if (mSearchLayout != null)
-      {
-        DisplayMetrics displayMetrics = new DisplayMetrics();
-        WindowManager windowmanager = (WindowManager) mFrame.getContext().getSystemService(Context.WINDOW_SERVICE);
-        windowmanager.getDefaultDisplay().getMetrics(displayMetrics);
-        // Get available screen height in DP
-        int height =  Math.round(displayMetrics.heightPixels / displayMetrics.density);
-        // If height is less than 400dp, the search wheel in a straight line
-        // In this case, move the pivot for the animation
-        if (height < 400)
-        {
-          UiUtils.waitLayout(mSearchLayout, () -> {
-            mSearchLayout.setPivotX(0);
-            mSearchLayout.setPivotY(mSearchLayout.getMeasuredHeight() / 2f);
-          });
-        }
-        for (SearchOption searchOption : SearchOption.values())
-          mFrame.findViewById(searchOption.mResId).setOnClickListener(this);
-      }
+      UiUtils.waitLayout(mSearchLayout, () -> {
+        mSearchLayout.setPivotX(0);
+        mSearchLayout.setPivotY(mSearchLayout.getMeasuredHeight() / 2f);
+      });
     }
-    return mSearchLayout;
+    for (SearchOption searchOption : SearchOption.values())
+      mFrame.findViewById(searchOption.mResId).setOnClickListener(this);
+    return true;
   }
 
   public void show(boolean show)
   {
     UiUtils.showIf(show, mSearchButton);
-    View searchLayout = getSearchLayout();
-    if (searchLayout != null)
-      UiUtils.showIf(show && mIsExpanded, searchLayout);
+    if (initSearchLayout() && mSearchLayout != null)
```

Comment:
mSearchLayout != null is not needed here and below. All checks should be done in initSearchLayout

Reply:
A solution would be to remove the `@Nullable` annotation from the `mSearchLayout` variable declaration. The editor would not complain and using only `if(initSearchLayout())` would not raise a warning. Which solution do you prefer?

Reply:
Well, I think that developers are smart enough to realize a logic in this simple case without IDE warnings :)
I'd prefer not to duplicate checks in code, so remove Nullable?